### PR TITLE
Add the new TimezoneType schema type.

### PR DESCRIPTION
**Pull request by ralt**

### CODE REVIEW

File: colander_tools/test_timezone.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+def test_timezone():
+    tz = timezone.TimezoneType().serialize({}, "Europe/Paris")
+    assert(tz == "Europe/Paris")
+
+    tz = timezone.TimezoneType().deserialize({}, "Europe/Paris")
+    normal = datetime.datetime(2009, 9, 1)
+
+    assert(tz.tzname(normal) == "CEST")
```
Comment (damz) [inline colander_tools/test_timezone.py:16]: Please add tests for the invalid cases.

File: colander_tools/timezone.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        if cstruct is colander.null:
+            return colander.null
+
+        try:
+            return pytz.timezone(cstruct)
+        except exceptions.UnknownTimeZoneError:
+            raise colander.Invalid(
+                node,
+                "`{}` is an invalid timezone".format(cstruct),
+            )
```
Comment (damz) [inline colander_tools/timezone.py:36]: Let's make this translatable while we are at it.

Comment (ralt) [inline colander_tools/timezone.py:36]: What do you mean?

Comment (damz) [inline colander_tools/timezone.py:36]: Like in https://github.com/platformsh/colander-tools/blob/master/colander_tools/strict.py#L22

**Comment by ralt** [comment]
Both points fixed -- I also made sure that `serialize()` throws an error for invalid values.

File: colander_tools/timezone.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        except exceptions.UnknownTimeZoneError:
+            raise colander.Invalid(
+                node,
+                colander._(
+                    '"${val}" is not a known timezone',
+                    mapping={"val": appstruct},
+                ),
+            )
+
+        return str(appstruct)
```
Comment (damz) [inline colander_tools/timezone.py]: Hm. That should be entitled to get a pytz tzinfo object as the `appstruct` no? If it is not a timezone object, just raise Invalid? If it is, return `appstruct.zone`.

Comment (ralt) [inline colander_tools/timezone.py]: Indeed -- I've updated the code to reflect this, as well as added a new test.

**Pull request merged**